File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-0.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
 
public class Solution {
    public ListNode MergeTwoLists(ListNode list1, ListNode list2)
    {
        ListNode head = new ListNode();
        ListNode l1 = list1;
        ListNode l2 = list2;
        ListNode curr = head;

        while(l1 != null && l2 != null)
        {

            if(l1.val <= l2.val)
            {
                curr.next = l1;
                l1 = l1.next;
                curr = curr.next;
            }
            else if(l2.val < l1.val)
            {
                curr.next = l2;
                l2 = l2.next;
                curr = curr.next;
            }   
        }

         
        if(l1 != null)
        {
            curr.next = l1;
        }

        if(l2 != null)
        {
            curr.next = l2;
        }

        return head.next;
    }
}

File: Data Structures & Algorithms/add-two-numbers/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:

        carry = 0;
        result =[]
        number1 = ""
        number2 = ""
        result = 0
        curr = l1;
        head = ListNode()

        while(curr != None ):
            number1+=str(curr.val)
            curr = curr.next;

        curr = l2;
        while(curr != None ):
            number2+=str(curr.val)
            curr = curr.next

        number1 = "".join(reversed(number1))
        number2 = "".join(reversed(number2))

        print(number1)
        print(number2)

        
        sum = int(number1) + int(number2)
        print(sum)
        r_sum = "".join(reversed(str(sum)))

        curr = head;
        for c in str(r_sum):
            print(curr.val)
            curr.next = ListNode(int(c))
            curr = curr.next

        head = head.next;

        return head;
            
            
           
        

        